### PR TITLE
Add pitch-7 as a valid rotation

### PR DESCRIPTION
**Pull request by peterbarker**
Replaces https://github.com/ArduPilot/ardupilot/pull/11389/commits

### CODE REVIEW

**Comment by khancyr** [comment]
you miss the math test ?

File: libraries/AP_Math/vector3.cpp [hunk 1]
```diff
@@ -244,6 +244,15 @@ void Vector3<T>::rotate(enum Rotation rotation)
         x = tmp;
         return;
     }
+    case ROTATION_PITCH_7: {
+        const float sin_pitch = 0.12186934340514748; // sinf(pitch);
```
Comment (khancyr) [inline libraries/AP_Math/vector3.cpp]: miss f qualifier ;-P

Comment (peterbarker) [reply libraries/AP_Math/vector3.cpp]: Fixed on original PR, thanks.

**Comment by peterbarker** [comment]
Not replacing original PR; force-pushed over it.

**Pull request closed without merging**